src/main/scala/workflow/WorkflowUtils.scala

```diff
@@ -0,0 +1,124 @@
+package workflow
+
+import scala.collection.mutable.ArrayBuffer
+
+
+object WorkflowUtils {
+  def instructionsToPipeline[A, B](instructions: Seq[Instruction]): Pipeline[A, B] = {
```

Comment:
One thing here - is the Pipeline -> Instruction -> Pipeline conversion process fully deterministic and symmetric? That is - if I fed a pipeline through the steps above would I end up with the same exact pipeline? Not that it has to be, but that might be a nice property to be able to rely on..